### ISSUE TITLE
OCI8 binding prepared statements

### ISSUE THREAD

**Issue by matesko** [doctrine/dbal#3722]
### BC Break Report

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 2.10

#### Summary

After update to 2.10 variable binding not working as before. Various errors are thrown, for example ORA-01036 on select statements, and ORA-01008 on update statements 

#### Previous behaviour

Working with no issues

#### Current behavior

query example:
```sql
update rates
set
    rate_fixing = :rate
where
    to_char(rate_date, 'DD.MM.YYYY') = :d
    and currency = :cur
```
execution code:
```php
         $r = $q->execute(
                    [
                        ':rate'     => round($t['rate'], 6),
                        ':d' => $t['date'],
                        ':cur'   => $t['currency'],
                    ]
                );
```
```log
request.CRITICAL: Uncaught PHP Exception Doctrine\DBAL\DBALException: "An exception occurred while executing 'update rates set     rate_fixing = :rate where     to_char(rate_date, 'DD.MM.YYYY') = :d and currency = :cur' with params [6.69515, "11.09.2019", "USD"]:  Notice: Undefined index: :rate" at G:\apache_root\s4\dan\vendor\doctrine\dbal\lib\Doctrine\DBAL\DBALException.php line 169 {"exception":"[object] (Doctrine\\DBAL\\DBALException(code: 0): An exception occurred while executing 'update rates\r\nset\r\n    rate_fixing = :rate\r\nwhere\r\n    to_char(rate_date, 'DD.MM.YYYY') = :d\r\n    and currency = :cur' with params [6.69515, \"11.09.2019\", \"USD\"]:\n\nNotice: Undefined index: :rate at G:\\apache_root\\s4\\dan\\vendor\\doctrine\\dbal\\lib\\Doctrine\\DBAL\\DBALException.php:169, ErrorException(code: 0): Notice: Undefined index: :rate at G:\\apache_root\\s4\\dan\\vendor\\doctrine\\dbal\\lib\\Doctrine\\DBAL\\Driver\\OCI8\\OCI8Statement.php:281)"} []

```

**Comment by matesko** [comment]
The culprit is commit 7bafa27cf7467031f35bfefc8eea8b0805f479bb (PHPStan Level 7 )

lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php
line 277
$column = $this->_paramMap[$column] ?? $column;

?? $column 

was removed

**Comment by morozov** [comment]
Fixed by #3738.
